pkg/reconciler/utils/pubsub/subscription.go

```diff
@@ -64,6 +66,20 @@ func (r *Reconciler) ReconcileSubscription(ctx context.Context, id string, subCo
 			}
 			return r.createSubscription(ctx, id, subConfig, obj, updater)
 		}
+		// Update the subscription config in case the retry or dead letter policy changed. A nil policy indicates no change.
+		if (subConfig.RetryPolicy != nil && !equality.Semantic.DeepEqual(config.RetryPolicy, subConfig.RetryPolicy)) ||
+			(subConfig.DeadLetterPolicy != nil && !equality.Semantic.DeepEqual(config.DeadLetterPolicy, subConfig.DeadLetterPolicy)) {
+			updateSubConfig := pubsub.SubscriptionConfigToUpdate{
+				RetryPolicy:      subConfig.RetryPolicy,
+				DeadLetterPolicy: subConfig.DeadLetterPolicy,
+			}
+			if _, err := sub.Update(ctx, updateSubConfig); err != nil {
+				updater.MarkSubscriptionFailed("SubscriptionConfigUpdateFailed", "Failed to update Pub/Sub subscription config: %v", err)
+				return nil, err
+			}
+			logger.Info("Updated PubSub subscription config", zap.String("name", sub.ID()))
```

Comment:
Might be useful to include the new config in this log.

Reply:
Good idea, I'll make a new good first issue about this